helpers/errorSchema.go

```diff
@@ -0,0 +1,12 @@
+package helpers
+
+import "github.com/gin-gonic/gin"
+
+func errorSchema(err error, msg string, details string, statusCode uint8) gin.H {
+	return gin.H{
+		"error":      err.Error(),
+		"message":    msg,
+		"detail":     details,
+		"statusCode": statusCode, // not sure if this should be done
```

Comment:
I would suggest to have something like
```
{
    "status": 400,
    "data": {},
    "error":{
        "message": "OAuthException",
        "details":"Missing redirect_uri parameter.",
        "error": "err.Error()"
    }
}
```

the `data` holds the information that is supposed to be returned on a sucessfull response, in case of failure it should be empty as in the above example

Reply:
Even Facebook follows such convention, hit the below URL to see
https://graph.facebook.com/oauth/access_token?client_id=foo&client_secret=bar&grant_type=baz

Reply:
Actually the map returned is as declared by @abhi-shek-raj, if he thinks that the one laid in your comment is better, I would be happy to change it accordingly

Reply:
cc: @abhishekraj272

Reply:
it was a sample, you can use the @uzaxirr's one